src/base_alloc/CMakeLists.txt

```diff
@@ -0,0 +1,37 @@
+# Copyright (C) 2024 Intel Corporation
+# Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+include(${UMF_CMAKE_SOURCE_DIR}/cmake/helpers.cmake)
+
+set(BA_SOURCES
+    base_alloc.c
+    base_alloc_os_mem.c
+    base_alloc_linear.c
+)
+
+set(BA_SOURCES_LINUX
+    base_alloc_linux.c
+)
+
+set(BA_SOURCES_WINDOWS
+    base_alloc_windows.c
+)
+
+if(LINUX)
+    set(BA_SOURCES ${BA_SOURCES} ${BA_SOURCES_LINUX})
+elseif(WINDOWS)
+    set(BA_SOURCES ${BA_SOURCES} ${BA_SOURCES_WINDOWS})
+elseif(MACOSX)
+    set(BA_SOURCES ${BA_SOURCES} ${BA_SOURCES_MACOSX})
+endif()
+
+if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+	add_compile_options(-DDEBUG)
```

Comment:
fix indentation, also - shouldn't this be added to top - level cmake?

Reply:
I think NDEBUG is always defined for release builds, so cannot we just use `#ifndef NDEBUG`?

Reply:
Done.